### ISSUE TITLE
VB tuples: crash in SubstituteNamedType.MakeDeclaredBase

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#13305]
See `MultipleDefinitionsOfValueTuple`

```VB

        <Fact(Skip:="REMOVE")>
        Public Sub MultipleDefinitionsOfValueTuple()

            Dim source1 =
<compilation name="comp1">
    <file name="a.vb">
Public Module M1
    &lt;System.Runtime.CompilerServices.Extension()&gt;
    Public Sub Extension(x As Integer, y As (Integer, Integer))
        System.Console.Write("M1.Extension")
    End Sub
End Module
<%= s_trivial2uple %></file>
</compilation>

            Dim source2 =
<compilation name="comp2">
    <file name="a.vb">
Public Module M2
    &lt;System.Runtime.CompilerServices.Extension()&gt;
    Public Sub Extension(x As Integer, y As (Integer, Integer))
        System.Console.Write("M2.Extension")
    End Sub
End Module
<%= s_trivial2uple %></file>
</compilation>

            Dim comp1 = CreateCompilationWithMscorlibAndVBRuntime(source1, additionalRefs:={MscorlibRef_v46})
            comp1.AssertNoDiagnostics()
            Dim comp2 = CreateCompilationWithMscorlibAndVBRuntime(source2, additionalRefs:={MscorlibRef_v46})
            comp2.AssertNoDiagnostics()

            Dim source =
<compilation name="comp">
    <file name="a.vb">
Imports System
Imports M1
Imports M2
Class C
    Public Shared Sub Main()
        Dim x As Integer = 0
        x.Extension((1, 1))
    End Sub
End Class
</file>
</compilation>

            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, additionalRefs:={comp1.ToMetadataReference(), comp2.ToMetadataReference()})
            comp.AssertTheseDiagnostics(
<errors>
</errors>)
            ' Crashes in SubstituteNamedType.MakeDeclaredBase
            ' REMOVE AND UPDATE AFTER REVIEW

            Dim verifier = CompileAndVerify(source, additionalRefs:={comp1.ToMetadataReference()}, expectedOutput:=<![CDATA[M1.Extension]]>)
            verifier.VerifyDiagnostics()

        End Sub

```

**Comment by jcouv** [comment]
Found a simpler repro, without extension methods:
```VB
        <Fact>
        Public Sub Crash2()

            Dim source1 =
<compilation name="comp1">
    <file name="a.vb">
Public Module M1
    Public Sub Method(x As Integer, y As System.ValueTuple(Of Integer, Integer))
        System.Console.Write("M1.Extension")
    End Sub
End Module
<%= s_trivial2uple %></file>
</compilation>


            Dim comp1 = CreateCompilationWithMscorlibAndVBRuntime(source1, additionalRefs:={MscorlibRef_v46})
            comp1.AssertNoDiagnostics()

            Dim source =
<compilation name="comp">
    <file name="a.vb">
Imports System
Imports M1
Class C
    Public Shared Sub Main()
        Dim x As Integer = 0
        M1.Method(0, new ValueTuple(Of Integer, Integer)(1, 2))
    End Sub
End Class
</file>
</compilation>

            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, additionalRefs:={comp1.ToMetadataReference()})
            comp.AssertTheseDiagnostics(
<errors>
</errors>)
            ' Crashes in SubstituteNamedType.MakeDeclaredBase

            Dim verifier = CompileAndVerify(source, additionalRefs:={comp1.ToMetadataReference()}, expectedOutput:=<![CDATA[M1.Extension]]>)
            verifier.VerifyDiagnostics()

        End Sub
```

FYI @VSadov

**Comment by jcouv** [comment]
Fixed by #13546
